src/libraries/filcrypto/filproofs/algorithms.go

```diff
@@ -266,62 +268,59 @@ func (sdr *WinStackedDRG_I) Seal(sid sector.SectorID, data []byte, randomness se
 		CommCTreePath_:     commCTreePath,
 		CommQTreePath_:     commQTreePath,
 		CommRLastTreePath_: commRLastTreePath,
-		Seeds_:             sealSeeds,
+		Seed_:              sealSeed,
 		KeyLayers_:         windowKeyLayers,
 		QLayer_:            qLayer,
 		Replica_:           replica,
 	}
 	return &result
 }
 
-func (sdr *WinStackedDRG_I) _generateWindowKey(windowIndex int, sid sector.SectorID, commD sector.UnsealedSectorCID, nodes int, randomness sector.SealRandomness) ([][]byte, sector.SealSeed) {
-	sealSeed := computeSealSeed(sid, windowIndex, randomness, commD)
+func (sdr *WinStackedDRG_I) _generateWindowKey(sealSeed sector.SealSeed, windowIndex int, sid sector.SectorID, commD sector.UnsealedSectorCID, nodes int, randomness sector.SealRandomness) [][]byte {
 	nodeSize := int(sdr.NodeSize())
 	curveModulus := sdr.Curve().FieldModulus()
 	layers := int(sdr.Layers())
 
 	keyLayers := generateSDRKeyLayers(sdr.Drg(), sdr.Expander(), sealSeed, windowIndex, nodes, layers, nodeSize, curveModulus)
 
-	return keyLayers, sealSeed
+	return keyLayers
 }
 
 func (sdr *WinStackedDRG_I) GenerateCommitments(replica []byte, windowKeyLayers [][]byte, qLayer []byte) (commC PedersenHash, commQ PedersenHash, commRLast PedersenHash, commR PedersenHash, commCTreePath file.Path, commQTreePath file.Path, commRLastTreePath file.Path) {
 	commC, commCTreePath = computeCommC(windowKeyLayers, int(sdr.NodeSize()))
 	commQ, commQTreePath = computeCommQ(qLayer, int(sdr.NodeSize()))
 	commRLast, commRLastTreePath = BuildTree_PedersenHash(replica)
-
 	commR = TernaryHash_PedersenHash(commC, commQ, commRLast)
 
-	// FIXME: need to compute commQ.
 	return commC, commQ, commRLast, commR, commCTreePath, commQTreePath, commRLastTreePath
 }
 
-func computeSealSeed(sid sector.SectorID, windowIndex int, randomness sector.SealRandomness, commD sector.UnsealedSectorCID) sector.SealSeed {
-	var proverId []byte // TODO: Derive this from sid.MinerID()
+func getProverID(minerID sector.MinerID) []byte {
+	panic("TODO")
+}
+func computeSealSeed(sid sector.SectorID, randomness sector.SealRandomness, commD sector.UnsealedSectorCID) sector.SealSeed {
+	proverId := getProverID(sid.MinerID())
 	sectorNumber := sid.Number()
 
 	var preimage []byte
 	preimage = append(preimage, proverId...)
 	preimage = append(preimage, littleEndianBytesFromUInt(UInt(sectorNumber), 8)...)
```

Comment:
Diff: the current implementation uses `BigEndian`

Reply:
Thanks, dig.

Reply:
Can you confirm that this applies everywhere we are constructing hash preimages? (We discussed, so I think so.)